### DISCUSSION TITLE
Use own LFS server for Github - No enterprise plan

### DISCUSSION THREAD

**Discussion by vdawg-git** [git-lfs/git-lfs#5583] [Q&A] [answered]
Hey, I want to create a website for an open-source software, where users can contribute showcases of the plugins they created for it. Those can include images and videos.

So using Git LFS seems like the perfect solution- but Github only allows that for public repos with the Enterprise plan.

Is it possible to instead self host an LFS server and configure the repo to use this one, so that Github will not prevent PRs to it? Or will Github block all PRs regardless? 

Thank you guys :)

**Comment by bk2204** [comment, marked as answer]
Sure, you can do this.  There's a [large number of implementations you can use](https://github.com/git-lfs/git-lfs/wiki/Implementations), including our own [LFS test server](https://github.com/git-lfs/lfs-test-server/) which, while limited, can be used in production as well if you like.

You'd simply create a Git-style configuration file in `.lfsconfig` in your repository and set `lfs.url` to the LFS URL.  You can look at the `git-lfs-config` manual page to see how to configure that.

**Reply by vdawg-git** [reply]
Perfect and thank you! :)

**Reply by tnguye40** [reply]
Hi @bk2204 

I successfully implemented `LFS test server` with my Github Enterprise Server as a remote LFS storage. 

After using `BFG Repo Cleaner` tool to purge a LFS file from the repository's history and executing the command `git lfs prune --force` I still notice that the LFS file is still cached in the remote LFS server.

It seems that Github ([https://github.com/git-lfs/git-lfs/wiki/Limitations](https://github.com/git-lfs/git-lfs/wiki/Limitations)) has no way of deleting a remote LFS object.

To free up some space, do I need to clean up manually the cache in the remote LFS server by trying to identify the file(s) associated to the removed LFS object ?

**Reply by bk2204** [reply]
The Git LFS protocol has no way to delete objects, so you'd need to write into GitHub support and ask them to delete the objects for you if you no longer want them.

**Reply by tnguye40** [reply]
Actually we have deployed a Github Enterprise instance with OVA virtual appliance provided by Github. 
I have deployed the third-party `LFS test server` in another VM to avoid to store the LFS files directly in the Github virtual appliance.

My question is to know if there is a (trivial) method to remove the files stored in the remote LFS storage when the referenced LFS object has been removed from the repository's history ? 

If the LFS file is stored directly within the Github virtual appliance, it is kept 3 months before definitive removal when the LFS object is unreferenced.

* https://docs.github.com/en/enterprise-server@3.11/repositories/working-with-files/managing-large-files/removing-files-from-git-large-file-storage#removing-a-single-file
* https://docs.github.com/en/enterprise-server@3.11/authentication/keeping-your-account-and-data-secure/removing-sensitive-data-from-a-repository
